client/web/src/insights/pages/dashboards/DasbhoardsRoutes.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react'
+import { Route, RouteComponentProps, Switch, useRouteMatch } from 'react-router'
+
+import { ExtensionsControllerProps } from '@sourcegraph/shared/src/extensions/controller'
+import { PlatformContextProps } from '@sourcegraph/shared/src/platform/context'
+import { SettingsCascadeProps } from '@sourcegraph/shared/src/settings/settings'
+import { TelemetryProps } from '@sourcegraph/shared/src/telemetry/telemetryService'
+
+import { AuthenticatedUser } from '../../../auth'
+import { Settings } from '../../../schema/settings.schema'
+
+import { InsightsDashboardCreationPage } from './creation/InsightsDashboardCreationPage'
+import { DashboardsPage } from './dashboard-page/DashboardsPage'
+
+export interface DashboardsRoutesProps
+    extends TelemetryProps,
+        SettingsCascadeProps<Settings>,
+        ExtensionsControllerProps,
+        PlatformContextProps<'updateSettings'> {
+    authenticatedUser: AuthenticatedUser
+}
+
+/**
+ * Displays Code Insights dashboard area.
+ */
+export const DashboardsRoutes: React.FunctionComponent<DashboardsRoutesProps> = props => {
+    const { authenticatedUser, settingsCascade, platformContext, telemetryService, extensionsController } = props
+    const match = useRouteMatch()
+
+    return (
+        <Switch>
+            <Route
+                path={`${match.url}/dashboard/:dashboardId?`}
+                render={(routeProps: RouteComponentProps<{ dashboardId: string }>) => (
+                    <DashboardsPage
+                        telemetryService={telemetryService}
+                        extensionsController={extensionsController}
+                        settingsCascade={settingsCascade}
+                        dashboardID={routeProps.match.params.dashboardId}
+                    />
+                )}
+            />
+
+            <Route
+                path={`${match.url}/add-dashboard`}
```

Comment:
The pattern we have currently in our app is using `/new` as a subroute of the collection, e.g. `/organizations/new` and here `/dashbodards/new`
```suggestion
                path={`${match.url}/dashboards/new`}
```
This may require putting this `<Route>` first so that it takes precedence over the generic dashboard route.

Reply:
@felixfbecker I took this URL from our figma designs.

I think @AlicjaSuska was trying to avoid overlapping between this `/dashboards/new` when we want to create a new dashboard and `/dashboards/new` when we have a dashboard that has `new `as a name